.vscode/tasks.json

```diff
@@ -1,46 +1,222 @@
+                "cwd": "${workspaceFolder}/src"
+            },
+            "args": [
+                "test"
             ],
+            "presentation": {
+                "focus": true,
+                "clear": true
+            },
             "problemMatcher": "$msCompile"
         },
+        // Cleans the build outputs of the 'minsk.sln' solution
         {
-            "label": "mc",
+            "label": "clean",
             "command": "dotnet",
-            "type": "shell",
+            "type": "process",
+            "group": "build",
+            "options": {
+                "cwd": "${workspaceFolder}/src"
+            },
+            "args": [
+                "clean"
+            ],
+            "presentation": {
+                "focus": true,
+                "clear": true
+            },
+            "problemMatcher": "$msCompile"
+        },
+        // Builds the specified sample project as selected by the user
+        {
+            "label": "build-msproj",
+            "command": "dotnet",
+            "type": "process",
+            "group": "build",
+            "options": {
+                "cwd": "${workspaceFolder}/samples"
+            },
+            "args": [
+                "build",
+                "${input:msproj}"
+            ],
+            "presentation": {
+                "focus": true,
+                "clear": true
+            },
+            "problemMatcher": {
+                "owner": "minsk",
+                "fileLocation": "absolute",
+                "pattern": [
+                    {
+                        "regexp": "^(.+)\\((\\d+,\\d+,\\d+,\\d+)\\):\\s+(.+)$",
+                        "file": 1,
+                        "location": 2,
+                        "message": 3
+                    }
+                ]
+            }
+        },
+        // Builds and Runs the specified sample project as selected by the user
+        {
+            "label": "run-msproj",
+            "command": "dotnet",
+            "type": "process",
+            "options": {
+                "cwd": "${workspaceFolder}/samples"
+            },
             "args": [
                 "run",
                 "--project",
-                "${workspaceFolder}/src/msc/msc.csproj",
+                "${input:msproj}"
+            ],
+            "presentation": {
+                "focus": true,
+                "clear": true
+            },
+            "problemMatcher": {
+                "owner": "minsk",
+                "fileLocation": "absolute",
+                "pattern": [
+                    {
+                        "regexp": "^(.+)\\((\\d+,\\d+,\\d+,\\d+)\\):\\s+(.+)$",
+                        "file": 1,
+                        "location": 2,
+                        "message": 3
+                    }
+                ]
+            }
+        },
+        // Builds and Runs the Minsk compiler with the currently opened file's folder path
+        {
+            "label": "msc",
```

Comment:
This is currently broken due to references not being included, so it emits _type not resolved_ errors. One way to solve it, is by supporting `.rsp` response files to take defaults from it when building by `msc` alone. It will also be helpful for `msi` too.